src/core/webidl.js

```diff
@@ -252,208 +89,138 @@ const operationNames = {};
 const idlPartials = {};
 
 // Takes the result of WebIDL2.parse(), an array of definitions.
-function makeMarkup(parse) {
-  const pre = document.createElement("pre");
-  pre.classList.add("def", "idl");
-  pre.innerHTML = parse.map(writeDefinition).join("");
-  return pre;
+function makeMarkup(parse, definitionMap, { suppressWarnings } = {}) {
+  return hyperHTML`<pre class="def idl">${webidl2writer.write(parse, {
```

Comment:
I think we should move the second argument object out into its own variable - it will make the code a little flatter (in terms of nested "}"s) and easier to maintain. We can also use  object shorthands for the functions, instead of the arrow functions.